core/src/io/anuke/mindustry/game/Gamemode.java

```diff
@@ -13,14 +13,15 @@
     }}),
     sandbox(() -> new Rules(){{
         infiniteResources = true;
-        waves = true;
+        waves = false;
```

Comment:
Why disable waves for sandbox? I realize that it's supposed to be "safe" with no waves, but considering players can control waves, there's no reason to disable them outright. This will just cause confusion; people will play sandbox expecting a button to run waves, not knowing that you need to go and configure custom game options.

Reply:
@Anuken Yes, I understand your way of thinking. But you can disable wave timer and enable infinite resources in survival mode. I wanted to make sandbox a freebuild mode with no objective. And survival's objective is to survive waves. Same like the old sandbox. If you wish I can do that your way, but I propose renaming sandbox to freebuild which would let peoples know that something changed. Let me know what do you think of that.

Reply:
> But you can disable wave timer and enable infinite resources in survival mode.
You can, but people who want waves in sandbox won't know that. Considering many people can't even find the "custom game" button, I can already see people asking to have waves back in sandbox.

> I wanted to make sandbox a freebuild mode with no objective.
Why? I don't see any reason to do this. Having a button to press which spawns waves in no way hurts the experience. In fact, many people playing sandbox are probably doing it to test the effectiveness of turrets against enemies, which is impossible in this modified sandbox mode. Every time someone would like to test waves, they would need to go in and open a new dialog to do so. This is not a good user experience.

> but I propose renaming sandbox to freebuild which would let peoples know that something changed.
If I keep waves in sandbox, nothing would have changed. It would work the same way it does in 3.5. Even then, that's *not* equivalent to the old freebuild. This would just cause confusion.

Reply:
Ok, you convinced me. Should I put wave timer and other waves buttons in custom rules screen when selected sandbox gamemode then?

Reply:
Yes.